rasa/shared/core/domain.py

```diff
@@ -619,6 +621,39 @@ def _initialize_forms(
             `FormAction` which is implemented in the Rasa SDK.
         """
         if isinstance(forms, dict):
+            # looping through all the available forms and checking if a
```

Comment:
This is getting quite big. I would extract a method which can handle this for a single form, and then iteratively apply it to each form.